File: example/src/test/java/specs/fl/Fl.java
package specs.fl;

import org.concordion.api.BeforeSpecification;
import org.junit.rules.TemporaryFolder;
import specs.Specs;

import java.io.IOException;

public class Fl extends Specs {
    private static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();

    @BeforeSpecification
    public static void beforeSpec() throws IOException {
        TEMP_FOLDER.create();
    }

    public String getDir() {
        return TEMP_FOLDER.getRoot().getPath();
    }

    public boolean addFile(String name) throws IOException {
        return TEMP_FOLDER.newFile(name).exists();
    }
}

File: example/src/test/kotlin/specs/fl/Fl.kt
package specs.fl

import org.concordion.api.BeforeSpecification
import org.junit.rules.TemporaryFolder
import specs.Specs

open class Fl : Specs() {
    val dir: String
        get() = TEMP_FOLDER.root.path

    fun addFile(name: String?): Boolean {
        return TEMP_FOLDER.newFile(name).exists()
    }

    companion object {
        private val TEMP_FOLDER = TemporaryFolder()

        @JvmStatic
        @BeforeSpecification
        fun beforeSpec() {
            TEMP_FOLDER.create()
        }
    }
}
